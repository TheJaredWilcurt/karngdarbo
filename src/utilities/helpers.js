const fs = nw.require('fs');
const path = nw.require('path');

const helpers = {
  setCurrentWorkingDirectory: function (dir) {
    dir = dir.trim();
    if (fs.existsSync(dir)) {
      nw.process.chdir(path.join(dir));
    }
  },
  setHtmlTagClasses: function (theme, useCustomScrollbars) {
    let customScrollbars = '';
    if (useCustomScrollbars) {
      customScrollbars = 'custom-scrollbars';
    }

    let classes = [
      theme,
      customScrollbars
    ].join(' ').trim();

    document.documentElement.className = classes;
  }
};

export default helpers;
