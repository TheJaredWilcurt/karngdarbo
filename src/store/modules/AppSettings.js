import helpers from '@/utilities/helpers.js';

const fs = nw.require('fs');
const path = nw.require('path');

const settingsFile = path.join(nw.App.dataPath, 'settings.json');

export const state = {
  customScrollbars: false,
  theme: ''
};

export const getters = {};

export const mutations = {
  setCustomScrollbars: function (state, bool) {
    state.customScrollbars = bool;
    helpers.setHtmlTagClasses(state.theme, bool);
  },
  setTheme: function (state, theme) {
    theme = theme || 'dark-mode';
    state.theme = theme;
    helpers.setHtmlTagClasses(theme, state.customScrollbars);
  }
};

export const actions = {
  deleteSettings: function () {
    try {
      if (fs.existsSync(settingsFile)) {
        fs.unlinkSync(settingsFile);
      }
    } catch (err) {
      // eslint-disable-next-line no-console
      console.log('Error deleting settings file', err);
    }
  },
  loadSettings: function (store) {
    store.commit('setAppLoading', true, { root: true });

    if (fs.existsSync(settingsFile)) {
      let settings = {};

      fs.readFile(settingsFile, function (err, data) {
        if (err) {
          store.commit('setAppError', 'Unable to load settings.\n' + err, { root: true });
        }

        if (data && typeof(data) === 'string') {
          try {
            settings = JSON.parse(data);
          } catch (error) {
            store.commit('setAppError', 'Error attempting to load settings.\n' + error, { root: true });
            return;
          }
        }

        if (Object.keys(settings).length) {
          // Apply Settings
          store.commit('setCustomScrollbars', settings.customScrollbars);
          store.commit('reposList/setReposList', settings.reposList, { root: true });
          store.commit('setTheme', settings.theme);
        }

        store.commit('setAppLoading', false, { root: true });
      });
    } else {
      store.commit('setAppLoading', false, { root: true });
    }
  },
  saveSettings: function (store) {
    // Grab Settings
    let settings = {
      customScrollbars: store.state.customScrollbars,
      reposList: store.rootState.reposList.reposList,
      theme: store.state.theme
    };
    settings = JSON.stringify(settings, null, 2);
    fs.writeFile(settingsFile, settings, function (err) {
      if (err) {
        store.commit('setAppError', 'There was an error saving your settings. ' + err);
      }
    });
  },
  setCustomScrollbarsAndSave: function (store, bool) {
    store.commit('setCustomScrollbars', bool);
    store.dispatch('saveSettings');
  },
  setThemeAndSave: function (store, theme) {
    store.commit('setTheme', theme);
    store.dispatch('saveSettings');
  }
};

export default {
  namespaced: true,
  state,
  getters,
  mutations,
  actions
};
