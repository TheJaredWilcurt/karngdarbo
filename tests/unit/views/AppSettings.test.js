import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex, { Store } from 'vuex';

import AppSettings from '@/views/AppSettings.vue';
import helpers from '../testHelpers.js';

describe('AppSettings.vue', () => {
  let store;
  let localVue;

  beforeEach(() => {
    localVue = createLocalVue();
    localVue.use(Vuex);
    global.nw = helpers.nw;

    store = new Store({
      state: {
        themes: []
      },
      getters: {},
      mutations: {},
      actions: {},
      modules: {
        appSettings: {
          namespaced: true,
          state: {
            customScrollbars: true,
            theme: ''
          }
        }
      }
    });
  });

  describe('Created', () => {
    test('Default snapshot', () => {
      const wrapper = shallowMount(AppSettings, {
        store,
        localVue
      });

      expect(wrapper.html())
        .toMatchSnapshot();
    });
  });
});
