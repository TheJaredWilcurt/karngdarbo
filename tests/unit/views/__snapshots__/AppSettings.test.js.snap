// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppSettings.vue Created Default snapshot 1`] = `
<div class="app-settings">
  <button>
    Get Branches
  </button>
  <input type="text" style="width: 100%;">
  <div>
    <strong>App Loading</strong>: <pre></pre>
  </div>
  <div>
    <strong>App Error</strong>: <pre></pre>
  </div>
  <div>
    <strong>Repo Path</strong>: <pre></pre>
  </div>
  <div>
    <strong>Branches</strong>: <pre></pre>
  </div>
  <div>
    <label>
      <input type="checkbox">
      <strong>Use styled scrollbars</strong>
    </label>
    <!---->
  </div>
  <div>
    <strong>Pick Theme</strong>:
    <select></select>
  </div>
</div>
`;
